hr_expense_cancel/models/hr_expense.py

```diff
@@ -9,13 +9,13 @@ class HrExpenseSheet(models.Model):
 
     def action_cancel(self):
         for sheet in self:
-            account_move = sheet.account_move_id
-            sheet.account_move_id = False
+            account_move = sheet.account_move_ids
+            sheet.account_move_ids = False
             payments = sheet.payment_ids.filtered(lambda line: line.state != "cancel")
             # case : cancel invoice from hr_expense
             self._remove_reconcile_hr_invoice(account_move)
             # If the sheet is paid then remove payments
-            if sheet.state == "done":
+            if sheet.state in ("done", "approve"):
                 if sheet.expense_line_ids[:1].payment_mode == "own_account":
                     self._remove_move_reconcile(payments, account_move)
```

Comment:
You're changing ``account_move_id`` single record to multiple ``account_move_ids``.
Have you checked for example `self._remove_move_reconcile(payments, account_move)` can take as an argument multiple moves?

Reply:
Hello @yankinmax 
I'm not changing the field `account_move_id`.
It was changed on odoo source when migrating to 17.0

Here if I'm not wrong : https://github.com/OCA/OCB/commit/68fbdc964038ef6a1cf0d3df773db101ca81794a#diff-42c75e4d29952aadf0abdd1bfe7e066bef3db1275c16d64dd3f3b96a332a2ff3